### PR TITLE
Create morevp-simplified.md

### PR DESCRIPTION
**Pull request by smartcontracts**
Adds a simplified explainer of More Viable Plasma that attempts to give more intuition for the basic components of the protocol.

The biggest noticeable change is that the exit game is significantly simplified. The original document considered the game in a setting where all inputs or outputs could participate simultaneously. This was a design optimization made specifically with the limited exit time of Plasma MVP in mind and hid much of the intuition behind the general exit game.

As a result, this document instead considers the game from the point of view of each input or output owner individually. I believe the "mental mapping" from the game described in this document to the game described in the original post is relatively clear, but I may be wrong.

### CODE REVIEW

File: plasma/plasma-mvp/morevp-simplified [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+More Viable Plasma (“MoreVP”) proposes an exit game that removes the need for confirmation signatures and ensures correct transaction processing order.
+
+### Basic Exit Game
+
+Under this new protocol, we allow owners of both the *inputs or outputs* to any transaction to attempt an exit. 
+
+If the owner of an input `in` to a transaction `tx` wants to exit, the owner must prove that:
+1. `tx` is correctly formed and properly signed.
+2. One of the inputs to `tx` has already been spent in some transaction other than `tx`.
```
Comment (pdobacz) [inline plasma/plasma-mvp/morevp-simplified]: something not right here, in (2.)

File: plasma/plasma-mvp/morevp-simplified [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+If the owner of an output `out` of a transaction `tx` wants to exit, the owner must prove that:
+1. `tx` is correctly formed and properly signed.
+2. None of the inputs to `tx` have already been spent in some transaction other than `tx`.
+3. `out` has not been spent.
+
+The first requirement in both games ensures that the transaction is correctly formed, input amounts are greater than or equal to outputs amounts, and signatures over the inputs come from their correct respective owners.
+
+The second requirement has something to do with a concept we call transaction “canonicity.” Simply defined, a transaction is “canonical” if none of the transaction’s inputs have been previously spent.
+
+If a transaction is correctly formed and signed, but the transaction *is not* canonical, then some input to that transaction has double-spent. However, it may be the case that not *all* of the inputs to the transaction are double spent - other than this (now invalid) transaction, these inputs are unspent. In this case, we should allow the owners of these inputs to retrieve their funds. The third requirement filters out any inputs that have been double spent.
```
Comment (pdobacz) [inline plasma/plasma-mvp/morevp-simplified]: > If a transaction is correctly formed and signed, but the transaction is not canonical, then some input to that transaction has double-spent

"...double-spent in a transaction that <<came earlier>> (or in stricter terms, was included earlier)" to be precise

Comment (pdobacz) [reply plasma/plasma-mvp/morevp-simplified]: same paragraph:

> other than this (now invalid) transaction

this bit seems off, what is the "now invalid transaction"? what definition of valid we have here?

EDIT: OK I see what's meant, still it took me some time to parse the English here

File: plasma/plasma-mvp/morevp-simplified [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+Obviously, we don’t want the operator’s exit to be processed before Bob’s exit. This will unfortunately be the case if we determine priority based on the age of the output. To solve this problem, we change introduce a slight change to how exit priority is handled.
+
+In the MoreVP exit game, within an exit on a transaction `t`, we calculate the priority every input or output of `t` based on the position (block number, transaction index, output index) of the youngest input to `t` (“youngest-input priority”).
+
+Now we can very easily see that, as long as Alice’s input was included in the Plasma chain before the invalid transaction, Bob’s exit will be processed first. It’s pretty easy to prove that this holds for *any* transaction, as long as all of a transaction’s inputs were included in the chain before the invalid transaction. In practice, this simply means that clients should never spend a UTXO if they haven’t validated the entire chain up (and including) the transaction in which that UTXO was created.
```
Comment (pdobacz) [inline plasma/plasma-mvp/morevp-simplified]: I find this new train of thought a little less clear. 

Before, it went more or less like this:
0/ we need multilateral transactions signed by foreign parties, e.g. atomic swaps
1/ because of that, we need inputs/outputs to be exitable in an independent manner, to allow foreign parties to fend against an attacking operator and themselves
2/ because of that, we introduce input-based priority, otherwise we have the problem noted with "force-on-chain" a while ago - "what priority to use when?"
3/ because of that, we need to involve piggybacking (owner's responsibility for particular input/output exiting) and canonicallity (to deal with transactions that don't get included in the child blocks or get included in withhold blocks), which is the new exit game

This new reasoning here looks like an inverted version of the original above (at least that's how I see the original reasoning) and right now I'm actually slightly confused - how did we end up with this protocol in the end? :)

File: plasma/plasma-mvp/morevp-simplified (hunk 1 above)
Comment (smartcontracts) [reply plasma/plasma-mvp/morevp-simplified]: "Already spent" here means:
- `tx` is not included in the Plasma chain and another spend exists.
- `tx` is included and another spend is included before it.

File: plasma/plasma-mvp/morevp-simplified (hunk 3 above)
Comment (smartcontracts) [reply plasma/plasma-mvp/morevp-simplified]: This document describes the game at the core of the original document.

"in-flight exit on a single tx + piggybacking" is effectively running this game in parallel for all owners of inputs and outputs simultaneously. In this sense, the original document described one specific implementation of the game described in this document.

This document wasn't intended to be a specification of a specific implementation, but a specification for the general overall protocol. The idea here is that we first define the game (this post), then we prove the game is correct, then we implement the game and prove the implementation actually implements the game.

File: plasma/plasma-mvp/morevp-simplified (hunk 1 above)
Comment (smartcontracts) [reply plasma/plasma-mvp/morevp-simplified]: Will clarify this.

Comment (pdobacz) [reply plasma/plasma-mvp/morevp-simplified]: OK got it now. It's just that this:

> owner of an input in to a transaction tx wants to exit, the owner must prove that: (...) One of the inputs to tx has already been spent in some transaction other than tx.

is slightly counterintuitive at first. Most likely that owner of an input doesn't know if they want to exit their input, they'll usually just see their spend withheld and start the in-flight exit, assuming that by default it will make the spend forced on rootchain. If they're out of luck, someone else will prove the latter part (non-canonicality) and make the input exitable - which is what makes perfect sense, according to our prior discussions.

File: plasma/plasma-mvp/morevp-simplified.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+## TL;DR
+
+This document provides a largely simplified specification for More Viable Plasma, a version of [Minimal Viable Plasma](https://ethresear.ch/t/minimal-viable-plasma/426) (“Plasma MVP”) that removes the need for confirmation signatures.
+
+## Background
+
+Transactions in Plasma MVP are processed in time-wise order.
+The “exit priority” of any UTXO is determined by the block in which the transaction was included, then the index of the transaction within that block, and finally the index of the output in the list of transaction outputs (“output-age priority”).
+Plasma MVP requires the use of confirmation signatures to prevent the operator from including a valid transaction time-wise after an invalid transaction.
```
Comment (pdobacz) [inline plasma/plasma-mvp/morevp-simplified.md:11]: Suggestion let's stop referencing the obsolete (soon-to-be-forgotten) confirm-sigs, and focus on the problem at hand (in-flight transactions exitability). If this is intended to become part of spec, confirm-sigs are just confusing and irritating to read about for bazilionth time

File: plasma/plasma-mvp/morevp-simplified.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+Although clients with “hot” wallets can quickly (and usually automatically) sign these secondary signatures, clients with “cold” wallets will face an annoying user-flow (e.g. clicking some button on a hardware wallet).  
+
+Confirmation signatures must also be published to the Plasma chain to avoid certain attack vectors.
+This significantly reduces the amount of block space available for transactions. 
+
+This motivates the search for an alternative exit game that removes the need for confirmation signatures and simultaneously ensures all valid UTXOs can be processed before all invalid UTXOs. 
+
+## More Viable Plasma
+
+More Viable Plasma (“MoreVP”) proposes an exit game that removes the need for confirmation signatures and ensures correct transaction processing order.
```
Comment (pdobacz) [inline plasma/plasma-mvp/morevp-simplified.md:25]: on the other hand, this description misses the essential setting of the exit game - tx is in-flight.

File: plasma/plasma-mvp/morevp-simplified.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+2. None of the inputs to `tx` have already been spent in some transaction other than `tx`.
+3. `out` has not been spent.
+
+The first requirement in both games ensures that the transaction is correctly formed (input amounts are greater than or equal to outputs amounts, referencing valid inputs) and signatures over the inputs come from their correct respective owners.
+
+The second requirement has something to do with a concept we call transaction “canonicity.”
+Simply defined, a transaction is “canonical” if none of the transaction’s inputs have been previously spent.
+Let's suppose there exist two transactions `tx` and `other_tx` such that `tx` and `other_tx` share at least one input.
+If `tx` *is* included in the Plasma chain, then `tx` is not canonical if `other_tx` is included before `tx` (in an earlier block or in the same block and an earlier transaction index).
+If `tx` is *not* included in the Plasma chain, then `tx` is not canonical if `other_tx` exists.
```
Comment (pdobacz) [inline plasma/plasma-mvp/morevp-simplified.md]: "exists" is too vague here, maybe "revealed" or "submitted on-root during the exit game", considering some strict definition of that?

File: plasma/plasma-mvp/morevp-simplified.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+Note, however, that the above exit game fails to provide safety if exit priority is still determined by the position of the transaction input or output in the Plasma chain (“output-age priority”). 
+
+To illustrate, imagine the following scenario:
+
+1. Alice sends transaction `tx` to Bob.
+2. The operator creates a block with invalid transaction `inv_tx`, and includes `tx` after `inv_tx`.
+3. Bob starts the exit game with `tx`.
+4. The operator starts the exit game with `inv_tx`.
+
+Because both `tx` and `inv_tx` appear to be properly formed and signed, both Bob and the operator can prove (1).
```
Comment (pdobacz) [inline plasma/plasma-mvp/morevp-simplified.md]: which 1/2/3 are you referring to - I'm guessing that one of the list at the beginning, but which?

File: plasma/plasma-mvp/morevp-simplified.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+Similarly, because none of the inputs to `tx` and `inv_tx` have been spent in some transaction other than `tx` and `inv_tx`, respectively, both can prove (2).
+Finally, because Bob and the operator have not spent their outputs, both can prove (3). 
+
+Obviously, we don’t want the operator’s exit to be processed before Bob’s exit.
+This will unfortunately be the case if we determine priority based on the age of the output.
+To solve this problem, we change introduce a slight change to how exit priority is handled.
+
+In the MoreVP exit game, within an exit on a transaction `t`, we calculate the priority every input or output of `t` based on the position (block number, transaction index, output index) of the youngest input to `t` (“youngest-input priority”).
+
+Now we can very easily see that, as long as Alice’s input was included in the Plasma chain before the invalid transaction, Bob’s exit will be processed first.
```
Comment (pdobacz) [inline plasma/plasma-mvp/morevp-simplified.md]: this isn't enough. The reader will ask what if inputs to `inv_tx` were included before inputs to `tx`. This particular notion of an invalid transaction must be defined and it must be explained, why it is the only way the operator can cheat Alice and Bob (how other attempts to cheat are thwarted).

File: plasma/plasma-mvp/morevp-simplified (hunk 3 above)
Comment (pdobacz) [reply plasma/plasma-mvp/morevp-simplified]: That wasn't exactly the point I'm making, but anyway I got a little more accustomed with the wording here and I'm more ok with it, I stand corrected.

Maybe it's worth an additional highlight, what's the main feature of the protocol - namely correct exitability of multilateral, in flight transactions, without any cooperation between owners involved

File: plasma/plasma-mvp/morevp-simplified.md (hunk 5 above)
Comment (smartcontracts) [reply plasma/plasma-mvp/morevp-simplified.md:25]: The point here is to give the intuition that the exit game is safe in all contexts, whether the transaction is in-flight or not.

File: plasma/plasma-mvp/morevp-simplified.md (hunk 7 above)
Comment (smartcontracts) [reply plasma/plasma-mvp/morevp-simplified.md]: Both are outputs playing the exit game, so both are referencing the requirements for owners of outputs. Will clarify.

File: plasma/plasma-mvp/morevp-simplified.md (hunk 4 above)
Comment (smartcontracts) [reply plasma/plasma-mvp/morevp-simplified.md:11]: The point of this section is to give background and to motivate the rest of the design. It's important that people understand not only the design but the thought processes that led to that design.

File: plasma/plasma-mvp/morevp-simplified.md (hunk 6 above)
Comment (smartcontracts) [reply plasma/plasma-mvp/morevp-simplified.md]: Sure.

File: plasma/plasma-mvp/morevp-simplified.md (hunk 5 above)
Comment (pdobacz) [reply plasma/plasma-mvp/morevp-simplified.md:25]: I get that, and that's a worthwhile statement to include! If you don't mention in-flightness at all, you're not making a case for the complicated protocol. Let me rephrase:

> essential setting of the exit game - tx **might be** in-flight.

File: plasma/plasma-mvp/morevp-simplified (hunk 1 above)
Comment (smartcontracts) [reply plasma/plasma-mvp/morevp-simplified]: Yeah I wanted to avoid bringing in all of the complexity of whether or not the owner knows if it's even possible to prove the input is non-canonical. 

In the ideal world (let's imagine a very long exit period and cheap gas), the following would happen:
- Owner of output to some in-flight transaction submits exit. Transaction is shown to be non-canonical and owner doesn't get to exit.
- Owner of input now knows in-flight transaction is non-canonical and submits exit on the input.

This way we avoid the specific complexity of Plasma MVP because we're forced to finish the game within a short period of time and with as little gas as possible.

File: plasma/plasma-mvp/morevp-simplified.md (hunk 5 above)
Comment (smartcontracts) [reply plasma/plasma-mvp/morevp-simplified.md:25]: Agreed, changed.

File: plasma/plasma-mvp/morevp-simplified (hunk 1 above)
Comment (pdobacz) [reply plasma/plasma-mvp/morevp-simplified]: ah, I see your point. Still, now, after all our clarifications, I find the two-phase, explicit descritption of the game (with piggybacking etc, or as you're calling it - the implementation of the game) more approachable and intuitive. It seems it might not be easy to describe / explain what's going on without these specifics and gory details
